webapp/src/containers/addOrViewComments/index.tsx

```diff
@@ -46,12 +46,22 @@ const AddOrViewComments = () => {
     }, []);
 
     const getCommentsPayload = (): CommentsPayload => {
-        const data = getGlobalModalState(pluginState).data as CommentAndStateModalData;
-        return {
-            record_type: data?.recordType,
-            record_id: data?.recordId,
-            comments,
+        const {data} = getGlobalModalState(pluginState);
+        let payload: CommentsPayload = {
+            record_type: '',
+            record_id: '',
         };
+
+        if (data) {
+            const {recordId, recordType} = getGlobalModalState(pluginState).data as CommentAndStateModalData;
+            payload = {
+                record_type: recordType,
+                record_id: recordId,
+                comments,
+            };
+        }
```

Comment:
What was wrong in the previous code? I think it was fine. You could have done this -
```
return {
            record_type: data?.recordType || '',
            record_id: data?.recordId || '',
            comments,	
        };
```